example/remove/t8_example_empty_trees.cxx

```diff
@@ -0,0 +1,189 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element types in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <t8.h>
+#include <t8_forest/t8_forest.h>
+#include <t8_forest/t8_forest_types.h>
+#include <t8_cmesh/t8_cmesh_examples.h>
+#include <t8_schemes/t8_default/t8_default_cxx.hxx>
+#include <sc_options.h>
+#include <string>
+
+T8_EXTERN_C_BEGIN ();
+
+/** Removes all elements of a local tree if they belong to the corresponding
+ *  global trees which is given by the user_data. */
+static int
+t8_adapt_remove (t8_forest_t forest,
+                 t8_forest_t forest_from,
+                 t8_locidx_t which_tree,
+                 t8_locidx_t lelement_id,
+                 t8_eclass_scheme_c *ts,
+                 const int is_family,
+                 const int num_elements, t8_element_t *elements[])
+{
+  const t8_locidx_t  *tree_id =
+    (const t8_locidx_t *) t8_forest_get_user_data (forest);
+  const t8_gloidx_t   global_tree_id =
+    t8_forest_global_tree_id (forest_from, which_tree);
+  if (global_tree_id == (t8_gloidx_t) *tree_id) {
+    return -2;
+  }
+  return 0;
+}
+
+void
+t8_strip_of_quads (t8_locidx_t num_trees, t8_locidx_t empty_tree,
```

Comment:
```suggestion
t8_strip_of_quads (const t8_gloidx_t num_trees, const t8_gloidx_t empty_tree,
```

Reply:
I changed the type to `t8_gloidx_t`, but they cannot be const since they are passed to functions which do require not-const arguments.